### PR TITLE
Fix `ErnieMEmbeddings` device issue

### PR DESCRIPTION
**Pull request by ydshieh**
# What does this PR do?

Fix `ErnieMEmbeddings` CI failure for 
```bash
tests/models/ernie_m/test_modeling_ernie_m.py::ErnieMModelTest::test_multi_gpu_data_parallel_forward
```

See comments in the changes.

### CODE REVIEW

File: src/transformers/models/ernie_m/modeling_ernie_m.py [hunk 1]
```diff
@@ -77,15 +77,14 @@ def forward(
             inputs_embeds = self.word_embeddings(input_ids)
         if position_ids is None:
             input_shape = inputs_embeds.size()[:-1]
-            ones = torch.ones(input_shape, dtype=torch.int64)
+            ones = torch.ones(input_shape, dtype=torch.int64, device=inputs_embeds.device)
```
Comment (ydshieh) [inline src/transformers/models/ernie_m/modeling_ernie_m.py:80]: we can rely on this `inputs_embeds`

File: src/transformers/models/ernie_m/modeling_ernie_m.py [hunk 2]
```diff
@@ -77,15 +77,14 @@ def forward(
...
             seq_length = torch.cumsum(ones, dim=1)
             position_ids = seq_length - ones
 
             if past_key_values_length > 0:
                 position_ids = position_ids + past_key_values_length
         # to mimic paddlenlp implementation
         position_ids += 2
-        position_ids = position_ids.to(next(self.position_embeddings.parameters()).device)
```
Comment (ydshieh) [inline src/transformers/models/ernie_m/modeling_ernie_m.py:88]: With `nn.DataParallel`, we get `E       StopIteration: Caught StopIteration in replica 1 on device 1.` See [this](https://github.com/huggingface/transformers/blob/03aaac35021dead4fb7ad354fe9c986d16869f03/src/transformers/modeling_utils.py#L150).



Let's avoid using this `next(self.position_embeddings.parameters()).device` if possible.

**Review by ydshieh — COMMENTED** [review]
See comments.

**Review by younesbelkada — APPROVED** [review]
Thanks for the clean fix! 💯

File: src/transformers/models/ernie_m/modeling_ernie_m.py (hunk 2 above)
Comment (sgugger) [reply src/transformers/models/ernie_m/modeling_ernie_m.py:88]: Yes, this is bad practice in general and will also not work with Accelerate big model inference.

**Review by sgugger — APPROVED** [review]
Thanks for fixing thi!

**Pull request merged**